packages/@lwc/ssr-compiler/src/shared.ts

```diff
@@ -60,12 +60,4 @@ export interface IHoistInstantiation {
 
 export type TransformOptions = Pick<TemplateCompilerConfig, 'name' | 'namespace'>;
 
-// This is a mostly-correct regular expression will only match if the entire string
-// provided is a valid ECMAScript identifier. Its imperfections lie in the fact that
-// it will match strings like "export" when "export" is actually a reserved keyword
-// and therefore not a valid identifier. When combined with a check against reserved
-// keywords, it is a reliable test for whether a provided string is a valid identifier.
-const imperfectIdentifierMatcher = /^[_$a-zA-Z\xA0-\uFFFF][_$a-zA-Z0-9\xA0-\uFFFF]*$/;
-
-export const isValidIdentifier = (str: string) =>
-    !reservedKeywords.has(str) && imperfectIdentifierMatcher.test(str);
+export const isValidIdentifier = (str: string) => isValidES3Identifier(str);
```

Comment:
If we're not adding any extra logic, we should probably remove this and just use the helper directly.

Reply:
Do you mean the [isIdentifierName](https://babeljs.io/docs/babel-helper-validator-identifier#isidentifiername) helper? I checked the implementation and it seemed a little different in terms of reserved names, etc.

Reply:
I think Will just means that we don't need to re-declare a `const` here – we can just use `isValidES3Identifier` directly.

BTW it'd be great to add a comment pointing to this code so it's clear why we are including a random Babel dep in our Meriyah code :) 

https://github.com/salesforce/lwc/blob/4845840a8f11a0f0334a7fb50dd82b4f7c2cbe22/packages/%40lwc/babel-plugin-component/src/decorators/wire/transform.ts#L61-L63

Reply:
BTW I was curious if the `reservedKeywords` actually matters here, and it looks like no. This is a valid object:

<details><summary>Click to see</summary>

```js
const obj = {
  NaN: 'foo',
  arguments: 'foo',
  break: 'foo',
  case: 'foo',
  catch: 'foo',
  class: 'foo',
  const: 'foo',
  continue: 'foo',
  debugger: 'foo',
  default: 'foo',
  delete: 'foo',
  do: 'foo',
  else: 'foo',
  enum: 'foo',
  eval: 'foo',
  export: 'foo',
  extends: 'foo',
  false: 'foo',
  finally: 'foo',
  for: 'foo',
  function: 'foo',
  if: 'foo',
  implements: 'foo',
  import: 'foo',
  in: 'foo',
  instanceof: 'foo',
  interface: 'foo',
  let: 'foo',
  new: 'foo',
  null: 'foo',
  package: 'foo',
  private: 'foo',
  protected: 'foo',
  public: 'foo',
  return: 'foo',
  static: 'foo',
  super: 'foo',
  switch: 'foo',
  this: 'foo',
  throw: 'foo',
  true: 'foo',
  try: 'foo',
  typeof: 'foo',
  undefined: 'foo',
  var: 'foo',
  void: 'foo',
  while: 'foo',
  with: 'foo',
  yield: 'foo',
}
```

</details>

Reply:
So we can remove `reservedKeywords` entirely since it's not used anywhere.

Reply:
Ahh sorry yes that sounds good, thank you both